go/chat/sync.go

```diff
@@ -458,11 +465,34 @@ func (s *Syncer) sync(ctx context.Context, cli chat1.RemoteInterface, uid gregor
 			}
 		}
 
+		// The idea here is to limit the amount of work we do with the background conversation loader
+		// on mobile. If we are on a cell connection, or if we just came into the foreground, then limit
+		// the number of conversations we queue up for background loading.
 		var queuedConvs, maxConvs int
-		state := s.G().MobileNetState.State()
-		if state.IsLimited() {
-			maxConvs = 3
+		pageBack := 3
+		num := 50
+		netState := s.G().MobileNetState.State()
+		state := s.G().MobileAppState.State()
+		if s.G().IsMobileAppType() {
+			maxConvs = s.maxConvLoads
+			num = 30
+			pageBack = 0
+			if netState.IsLimited() || state == keybase1.MobileAppState_FOREGROUND {
```

Comment:
how do we know we just came into the foreground here?

Reply:
Just the foreground check is good enough